### PR TITLE
setLatLng constrained documentation + expose LatLngZoom

### LINKED ISSUE #10928: setZoom  overwrites setLatLng
Hi,

I've found that when calling setLatLng and then setZoom. The position set in setLatLng is not centered anymore. When calling setZoom first it works as expected. It would be nice if this bug could be documented in the javadoc of [MapboxMap](https://www.mapbox.com/android-docs/api/map-sdk/5.3.1/com/mapbox/mapboxsdk/maps/MapboxMap.html#setZoom-double-).

**Platform: Android**
**Mapbox SDK version: 5.3.1**

### Steps to trigger behavior

 1. Call MapboxMap.setLatLng(position)
 2. Call MapboxMap.setZoom(zoom)

### Expected behavior
The mapview shows the map in the set zoom and at the position centered.

### Actual behavior
The mapview shows the map in the set zoom and at some location near the position.

PS: I've seen this issue on the OnePlus 5 and OnePlus 5t. I've also seen it on an emulator of a colleague but was not able to replicate it by myself. Also my Motorola G 2014 (titan) wasn't showing bug.

### PR DESCRIPTION
**Pull request by tobrun**
Closes #10928

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java [hunk 1]
```diff
@@ -640,6 +643,21 @@ public void setZoom(@FloatRange(from = MapboxConstants.MINIMUM_ZOOM, to = Mapbox
...
 
+  /**
+   * Moves the center and the zoom of the camera specified by a LatLng object and double zoom.
+   * <p>
+   * Note that at low zoom levels, setLatLng is constrained so that the entire viewport shows map data.
+   * </p>
+   *
+   * @param latLng Target location to change to
+   */
+  public void setLatLngZoom(@NonNull LatLng latLng,
```
Comment (jfirebaugh) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java]: I don't think we should add this. It's already possible to set center and zoom simultaneously with `moveCamera` -- there's no need to add to what is already a sprawling camera API. If anything, we should be _removing_ individual setters such as `setLatLng` and `setZoom` in favor of the unified `moveCamera`.

Comment (tobrun) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java]: I initially implemented that api for having a more performant way of doing camera updates with Android SDK animators. After running benchmarks on both setups, the difference is neglectable:

screen shot 2018-02-14 at 07 34 32

thank you for flagging @jfirebaugh, I will work on removing the API.

Comment (LukasPaczos) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java]: @tobrun I'd be great to sync the removal of exposed setters with https://github.com/mapbox/mapbox-plugins-android/issues/259 and https://github.com/mapbox/mapbox-gl-native/issues/11170 as right now exposed setters are the only way to manipulate camera without a loss of tracking mode.

Comment (tobrun) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java]: happening in https://github.com/mapbox/mapbox-gl-native/pull/11201

**Pull request merged**